support-frontend/app/controllers/RegularContributions.scala

```diff
@@ -88,9 +88,9 @@ class RegularContributions(
         InternalServerError
       },
       { statusResponse =>
-        if (!testUsers.isTestUser(request)) {
+        if (!testUsers.isTestUser(request) && body.paymentFields.isDefined) {
```

Comment:
why not just pattern match and use the `aaa` below?

```suggestion
        body.paymentFields match { case Some(aaa) if !testUsers.isTestUser(request) =>
```